stdlib/string.gr

```diff
@@ -2016,6 +2016,148 @@ provide let forEachCodePointi = (fn: (Number, Number) -> Void, str: String) => {
   void
 }
 
+/**
+ * Iterates over Unicode characters in a string.
+ *
+ * @param fn: The iterator function
+ * @param str: The string to iterate
+ *
+ * @example String.forEachChar(print, "Hello world")
+ *
+ * @since v0.6.0
+ */
+@unsafe
+provide let forEachChar = (fn: Char -> Void, str: String) => {
```

Comment:
You could probably share some code between this function and `forEachCodePoint`; same with `forEachChari`

Reply:
I made the `idx` change, but I dont think it is optimial to share code here, the type signatures of `fn` are different so its not as simple as just having a bool that says to use char or code point. meaning you would need to use a wrapper function into the helper that tags the char, which it invokes an extra function call. On another note if we do want to go this route we might as well justhave a `forEachCodePointIHelper` which returns an unsafe code point. and then each function just becomes `forEachCodePointIHelper((c, _) => fn(tagChar(c)), str)`.